### PR TITLE
Don't fetch edits on unsupported sites

### PR DESCRIPTION
**Pull request by yayuyokitano**
Should reduce memory usage a lot for people with many tabs and many saved edits.

EDIT: also checks for frames when allframes is false now, shouldve been added a long time ago

### CODE REVIEW

File: src/core/content/main.ts [hunk 1]
```diff
@@ -25,15 +27,23 @@ async function main() {
  */
 async function fetchConnector(): Promise<void> {
 	const connector = await getConnectorByUrl(window.location.href);
+	console.log('bbbbb', connector, window.location.href);
```
Comment (inverse) [inline src/core/content/main.ts:30]:
```suggestion
```

Lets remove this :)

File: src/core/content/main.ts [hunk 2]
```diff
@@ -25,15 +27,23 @@ async function main() {
...
 	if (!connector) {
 		return;
 	}
 
+	// Don't run the connector in frames if it's not allowed to run in frames
+	if (window !== top && !connector.allFrames) {
```
Comment (inverse) [inline src/core/content/main.ts:36]: What does the first conditional do? Not familiar with this :)

File: src/core/storage/saved-edits.ts [hunk 3]
```diff
@@ -12,8 +12,8 @@ class SavedEditsImpl extends SavedEditsModel {
 		BrowserStorage.LOCAL_CACHE
 	);
 
-	constructor() {
-		super(BrowserStorage.getLocalStorage(BrowserStorage.LOCAL_CACHE));
+	public init() {
+		this._init(BrowserStorage.getLocalStorage(BrowserStorage.LOCAL_CACHE));
```
Comment (inverse) [inline src/core/storage/saved-edits.ts:16]: Should we protect if it's already been init?

**Review by inverse — APPROVED** [review]

File: src/core/content/main.ts (hunk 2 above)
Comment (yayuyokitano) [reply src/core/content/main.ts:35]: window === top if content script is not running in a frame/iframe

File: src/core/storage/saved-edits.ts (hunk 3 above)
Comment (yayuyokitano) [reply src/core/storage/saved-edits.ts:16]: There shouldn't be any harm to running init twice except that you initialize the storage twice.
Though it is only ever run at the very beginning.
_init is protected.

**Pull request merged**